service/request.go

```diff
@@ -137,8 +138,34 @@ func (r *Request) BoolFromQuery(param string, w *ResponseWriter) (bool, bool) {
 	return b, true
 }
 
-func (r *Request) Unmarshal(w *ResponseWriter, body interface{}) bool {
-	if err := json.NewDecoder(r.Body).Decode(body); err != nil {
+func (r *Request) Unmarshal(w *ResponseWriter, body interface{}, templates ...interface{}) bool {
+	typ := r.Header.Get("Content-Type")
+	if typ == "" || typ == "application/x-www-form-urlencoded" {
+		// If Content-Type is set or is a form (probably set from curl), assume
+		// json.
+		typ = api.MediaTypeJSON
```

Comment:
I think we should use `curl --data-binary` instead of ignoring `application/x-www-form-urlencoded` like this. 
```suggestion
	if typ == "" {
		typ = api.MediaTypeJSON
```